src/aics-image-viewer/components/App/index.tsx

```diff
@@ -302,10 +304,14 @@ const App: React.FC<AppProps> = (props) => {
     return (settings || getChannelSettings()).find((channel) => channel.name === channelName);
   };
 
+  const setAllChannelsUnloaded = (numberOfChannels: number): void => {
+    setChannelVersions(new Array(numberOfChannels).fill(0));
+  };
+
   const setOneChannelLoaded = (index: number): void => {
-    const newLoadedChannels = getLoadedChannels().slice();
-    newLoadedChannels[index] = true;
-    setLoadedChannels(newLoadedChannels);
+    const newVersions = getChannelVersions().slice();
+    newVersions[index]++;
+    setChannelVersions(newVersions);
```

Comment:
I expected to find some code that says "if the version that just arrived is behind the current version, then don't set this channel data".  Is that implicit somewhere or am I misunderstanding how version works here?

Reply:
`version` is really behaving like an extension of the boolean field it replaced here: it's still getting set to "false" (0) when loading begins, then "true" (1) when data arrives, but any more recent requests that began between those two events will still have the headroom to change the value (to... extra-true?) and trigger an update. It's not thinking at all about which request is more recent. We probably ought to track the order of requests, but I was nervous about giving this array of numbers that responsibility, since its length may change with number of channels

Reply:
Actually the better thing to do may be
- keep a request counter in a ref
- on requests, increment the counter and have the channel loaded callback close around the value when it's created
- when the callback runs, check the value against the ref
- this keeps channel loading/loaded state certain, so we can return to booleans here

Reply:
I think I get it.  I'm not sure we NEED to account for async requests arriving out of order, but it is definitely possible for that to happen.